src/pybind/mgr/telegraf/module.py

```diff
@@ -292,9 +283,10 @@ def handle_command(self, cmd):
                 "Command not found '{0}'".format(cmd['prefix']))
 
     def self_test(self):
-        measurements = self.gather_measurements()
-        if len(measurements) == 0:
-            raise RuntimeError('No measurements found')
+    try:
+        next(self.gather_measurements())
```

Comment:
Calling next() here will only result in evaluating the first element (I think?) -- maybe this should be list() instead to ensure hitting all the entries in the iterators is working